### PR TITLE
KAFKA-3058: remove the usage of deprecated config properties

### PR DESCRIPTION
**Pull request by konradkalita**

### CODE REVIEW

File: core/src/main/scala/kafka/tools/MirrorMaker.scala [hunk 1]
```diff
@@ -192,7 +192,7 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       val producerProps = Utils.loadProps(options.valueOf(producerConfigOpt))
       // Defaults to no data loss settings.
       maybeSetDefaultProperty(producerProps, ProducerConfig.RETRIES_CONFIG, Int.MaxValue.toString)
-      maybeSetDefaultProperty(producerProps, ProducerConfig.BLOCK_ON_BUFFER_FULL_CONFIG, "true")
+      maybeSetDefaultProperty(producerProps, ProducerConfig.MAX_BLOCK_MS_CONFIG, Long.MaxValue.toString)
```
Comment (ijuma) [inline core/src/main/scala/kafka/tools/MirrorMaker.scala:195]: Please update the class scaladoc, after:

"For mirror maker, the following settings are set by default to make sure there is no data loss:"

File: core/src/main/scala/kafka/tools/MirrorMaker.scala [hunk 2]
```diff
@@ -54,7 +54,7 @@ import scala.util.control.ControlThrowable
  *       1. use new producer with following settings
  *            acks=all
  *            retries=max integer
- *            block.on.buffer.full=true
+ *            max.block.ms.config=max long
```
Comment (ijuma) [inline core/src/main/scala/kafka/tools/MirrorMaker.scala]: Thanks for adding this. I would write the actual config name, not the variable name. Something like:

`max.block.ms=max long`

(edited the value to be consistent with `retries` a few lines above)

**Comment by ijuma** [comment]
LGTM apart from minor comment.

File: core/src/main/scala/kafka/tools/MirrorMaker.scala (hunk 2 above)
Comment (konradkalita) [inline core/src/main/scala/kafka/tools/MirrorMaker.scala]: I fixed that, but I  see another inconsistency in naming. Shouldn't `MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION` in org.apache.kafka.clients.producer.ProducerConfig
be named
`MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION_CONFIG` ?

Comment (ijuma) [inline core/src/main/scala/kafka/tools/MirrorMaker.scala]: Thanks. Yes, but those variables are part of the public API and we cannot remove them without going through a deprecation cycle. It would have to be done in a separate PR and I'm not sure what is the current policy with regards to things like that (if we think it's worth doing or not).

**Comment by junrao** [comment]
Thanks for the patch. LGTM.
